merge.go

```diff
@@ -23,7 +24,55 @@ const (
 //
 // Merge operation maybe a very time-consuming operation when the database is large.
 // So it is recommended to perform this operation when the database is idle.
-func (db *DB) Merge() error {
+// If reopenAfterDone is true, the original file will be replaced by the merge file,
+// and db's index will be rebuilt after the merge is complete.
+func (db *DB) Merge(reopenAfterDone bool) error {
+	if err := db.merge(); err != nil {
+		return err
+	}
+
+	if !reopenAfterDone {
+		return nil
+	}
+
+	db.mu.Lock()
+	defer db.mu.Unlock()
```

Comment:
The following part is the same with Open, can we just close the old db and Open a new one to replace the current db?

```
db.Close
newdb, err := Open()
db = newdb
```

Reply:
ok

Reply:
There is a bit difference:
- Some unnecessary operations e.g. close hint file and release file lock.
- If there is writing under concurrent conditions, an error will be reported that the database is closed

I can extract the same part in open() into a function, what do you think?

Reply:
OK, extracting some common operations from Open will be better, then loading files and building indexes in this new function. And Open can call it too.

Reply:
ok, I have finished.